circleguard/investigator.py

```diff
@@ -226,42 +236,137 @@ def median_frametime(frametimes):
         return np.median(frametimes)
 
     @staticmethod
-    def _parse_beatmap(beatmap):
+    def _parse_beatmap(beatmap, easy, hard_rock):
         hitobjs = []
 
         # parse hitobj
-        for hit in beatmap.hit_objects_no_spinners:
+        for hit in beatmap.hit_objects(easy=easy, hard_rock=hard_rock):
+            time = hit.time.total_seconds() * 1000
+            end_time = time
+            obj_type = 0
+            # object end time is different depending on the type of object
+            # type of hit object is necessary for notelock calculations (0 circle, 1 slider, 2 spinner)
+            if hasattr(hit, "end_time"):
+                end_time = hit.end_time.total_seconds() * 1000
+                if hasattr(hit, "curve"):
+                    obj_type = 1
+                else:
+                    obj_type = 2
             p = hit.position
-            hitobjs.append([hit.time.total_seconds() * 1000, p.x, p.y])
+            hitobjs.append([time, [p.x, p.y], end_time, obj_type])
```

Comment:
This should really become its own class, too much magic indexing going on where I don't know what `hitobjs[i][0]` means. `hitobjs[i].time` is clearer

Reply:
I think I'll turn it into a list of hitobjects instead, since this part is basically removing attributes of hitobject that I'm not using

Reply:
yep that's better